### PR TITLE
Update release notes draft to version v1.28.0-rc.0

### PR DESCRIPTION
**Pull request by rashansmith**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the Release Notes Draft to k/k v1.28.0-rc.0

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
This is an automated PR generated from `krel The Kubernetes Release Toolbox`

### CODE REVIEW

**Review by AnaMMedina21 — APPROVED** [review]
LGTM 🚀

File: releases/release-1.28/release-notes/maps/pr-118828-map.yaml
Comment (sanchita-07) [inline releases/release-1.28/release-notes/maps/pr-118828-map.yaml:1]: We can delete that extra space.

**Review by sanchita-07 — COMMENTED** [review]
Looks good to me. Thanks @rashansmith

File: releases/release-1.28/release-notes/maps/pr-118828-map.yaml
Comment (rashansmith) [reply releases/release-1.28/release-notes/maps/pr-118828-map.yaml:1]: @sanchita-07 done!

**Review by sanchita-07 — APPROVED** [review]
/lgtm

**Comment by sanchita-07** [comment]
Needs approval :)
cc @kubernetes/release-team-leads

**Review by saschagrunert — APPROVED** [review]

**Comment by sanchita-07** [comment]
Thanks @saschagrunert :heart:

**Pull request merged**